lms/resources/_js_config.py

```diff
@@ -93,14 +94,42 @@ def add_document_url(self, document_url):
         self.config["urls"]["via_url"] = via_url(self._request, document_url)
         self._add_canvas_submission_params(document_url=document_url)
 
-    def enable_content_item_selection_mode(self):
+    def enable_content_item_selection_mode(self, form_action, form_fields):
         """
         Put the JavaScript code into "content item selection" mode.
 
         This mode shows teachers an assignment configuration UI where they can
         choose the document to be annotated for the assignment.
         """
-        self.config["mode"] = "content-item-selection"
+        self.config.update(
+            {
+                "mode": "content-item-selection",
+                "enableLmsFilePicker": False,
+                "formAction": form_action,
+                "formFields": form_fields,
+                "googleClientId": self._request.registry.settings["google_client_id"],
+                "googleDeveloperKey": self._request.registry.settings[
+                    "google_developer_key"
+                ],
+                # Pass the URL of the LMS that is launching us to our JavaScript code.
+                # When we're being launched in an iframe within the LMS our JavaScript
+                # needs to pass this URL (which is the URL of the top-most page) to Google
+                # Picker, otherwise Picker refuses to launch inside an iframe.
+                "customCanvasApiDomain": self._context.custom_canvas_api_domain,
+                "lmsUrl": self._context.lms_url,
+                # The "content item selection" that we submit to Canvas's
+                # content_item_return_url is actually an LTI launch URL with
+                # the selected document URL or file_id as a query parameter. To
+                # construct these launch URLs our JavaScript code needs the
+                # base URL of our LTI launch endpoint.
+                "ltiLaunchUrl": self._request.route_url("lti_launches"),
```

Comment:
It would be good to break up the items which are:

 * For canvas only
 * For google only
 * Shared

For example, we set `courseId` as it's only needed for Canvas below, but it would seem there's no reason to set `ltiLaunchUrl` if we aren't going to enable it.

I think a good end state would be to have `google.clientId` and `google.developerKey` as a nested dict, and the same for Canvas, but this would disrupt the tests and make this a very big diff. In the mean time, we could set the stage by building the dict in three parts. Intialise with the common bits, update with the Google bits and then optionally update with the Canvas bits.

Reply:
BTW. Good opportunity here to make this easier to understand by renaming the vars later. A lot of these option names are really hostile to understanding what it is.

Reply:
Gonna get the whole lot in one place in `JSConfig` first, and then take a lot at reorganising it (if we have time).

With things like `ltiLaunchUrl` I'm thinking it's simpler to just set them all the time, in all LMS's and all types of assignment, even if they're not used all the time. It's just the base LTI launch URL, so it's at least potentially a setting of general interest. If we ever need to use it in a new circumstance, it'll be there. But mostly it's just less logic, less variation, and less tests in the `JSConfig` code.

Reply:
Nested dicts within the `JSConfig` dict does seem like the direction to go in, yes. Once it's all in one place it'll hopefully be clear _which_ nested dicts would work best

Reply:
Yep, one for later.